x/participation/simulation/simulation.go

```diff
@@ -0,0 +1,248 @@
+package simulation
+
+import (
+	"math/rand"
+	"time"
+
+	simappparams "github.com/cosmos/cosmos-sdk/simapp/params"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+	fundraisingkeeper "github.com/tendermint/fundraising/x/fundraising/keeper"
+	fundraisingtypes "github.com/tendermint/fundraising/x/fundraising/types"
+
+	"github.com/tendermint/spn/testutil/sample"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+
+	"github.com/tendermint/spn/x/participation/keeper"
+	"github.com/tendermint/spn/x/participation/types"
+)
+
+func RandomAccWithBalance(ctx sdk.Context, r *rand.Rand,
+	bk bankkeeper.Keeper,
+	accs []simtypes.Account,
+	desired sdk.Coins,
+) (simtypes.Account, sdk.Coins, bool) {
+	// Randomize the set
+	r.Shuffle(len(accs), func(i, j int) {
+		accs[i], accs[j] = accs[j], accs[i]
+	})
+
+	for _, acc := range accs {
+		balances := bk.GetAllBalances(ctx, acc.Address)
+		if balances.IsAllGT(desired) {
+			return acc, balances, true
+		}
+	}
+
+	return simtypes.Account{}, sdk.NewCoins(), false
+}
+
+func RandomAuction(ctx sdk.Context, r *rand.Rand, fk fundraisingkeeper.Keeper) (auction fundraisingtypes.AuctionI, found bool) {
```

Comment:
We will need to test these methods. If not implemented in this PR we must create an issue

Reply:
We can test functions like `FindAccount`, but I'm not sure how we can appropriately test the proper functionality of these `Random*` functions since the outcome should be random.

Reply:
We can have test that FindRandom<Something> returns a value if one something has been created like we have for `x/launch/simulation/store_test.go`